From 924f328df429d619bf41eae046f57b664f9e4f05 Mon Sep 17 00:00:00 2001
From: Ilan Peer <ilan.peer@intel.com>
Date: Fri, 22 Dec 2023 11:04:23 +0200
Subject: [PATCH] tests: Extend the HE regulatory test to also exemplify SP AP

Extend he_6ghz_reg to also show how to configure SP AP with EIRP TX
power limit.

Signed-off-by: Ilan Peer <ilan.peer@intel.com>
---
 tests/hwsim/test_he.py | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/tests/hwsim/test_he.py b/tests/hwsim/test_he.py
index 5ad6fb59a..23013e9f7 100644
--- a/tests/hwsim/test_he.py
+++ b/tests/hwsim/test_he.py
@@ -1702,6 +1702,20 @@ def test_he_6ghz_reg(dev, apdev):
 
         # Allow few more Beacon frames
         time.sleep(0.5)
+
+        # Modify the regulatory power type to SP and provide the client EIRP
+        # limit
+        # EIRP = PSD + 10 * log(channel width)
+        # 16 = 3 + 10 * log(20)
+        hapd.set("vendor_elements", "")
+        hapd.set("he_6ghz_reg_pwr_type", "1")
+        hapd.set("reg_def_cli_eirp", "14")
+
+        if "OK" not in hapd.request("UPDATE_BEACON"):
+            raise Exception("UPDATE_BEACON failed")
+
+        # Allow few more Beacon frames
+        time.sleep(0.5)
     except Exception as e:
         if isinstance(e, Exception) and str(e) == "AP startup failed":
             if not he_supported():
-- 
2.43.0

